File: exercises/day23-Portal-Stabilization/java/src/main/java/craft/portal/TemporalAnomaly.java
package craft.portal;

import java.util.UUID;

public record TemporalAnomaly(
    String id,
    long timestamp,
    String dimension,
    int karmaDebt,
    String karmaSignature
) {
    public static TemporalAnomaly create(long timestamp, String dimension, int karmaDebt) {
        String karmaSignature = generateKarmaSignature(timestamp, dimension, karmaDebt);
        return new TemporalAnomaly(
            UUID.randomUUID().toString(),
            timestamp,
            dimension,
            karmaDebt,
            karmaSignature
        );
    }
    
    private static String generateKarmaSignature(long timestamp, String dimension, int karmaDebt) {
        return String.format("%s-%d-%d", dimension, timestamp, karmaDebt);
    }
    
    public boolean hasSameKarmaAs(TemporalAnomaly other) {
        return this.karmaSignature.equals(other.karmaSignature);
    }
}

File: exercises/day23-Portal-Stabilization/csharp/PortalStabilization/Models/TemporalAnomaly.cs
namespace PortalStabilization.Models;

public record TemporalAnomaly(
    string Id,
    long Timestamp,
    string Dimension,
    int KarmaDebt,
    string KarmaSignature)
{
    public static TemporalAnomaly Create(long timestamp, string dimension, int karmaDebt)
    {
        var karmaSignature = GenerateKarmaSignature(timestamp, dimension, karmaDebt);
        return new TemporalAnomaly(
            Guid.NewGuid().ToString(),
            timestamp,
            dimension,
            karmaDebt,
            karmaSignature
        );
    }
    
    private static string GenerateKarmaSignature(long timestamp, string dimension, int karmaDebt)
    {
        return $"{dimension}-{timestamp}-{karmaDebt}";
    }
    
    public bool HasSameKarmaAs(TemporalAnomaly other)
    {
        return KarmaSignature == other.KarmaSignature;
    }
}
